src/server/object_services/object_server.js

```diff
@@ -1974,7 +1977,14 @@ async function update_endpoint_stats(req) {
 /**
  * @param {nb.ObjectMD} obj
  */
-async function _complete_object_parts(obj) {
+async function _complete_object_parts(obj, num_parts) {
+
+    if (num_parts === 1) {
+        //skip db access if only one part
+        //(this part is already ok in db, no need to fix it)
```

Comment:
Is a single part different than any other number of parts?
The only difference is if we are on multipart where we need to fix the offsets.